packages/gatsby/src/datastore/lmdb/query/__tests__/suggest-index.ts

```diff
@@ -0,0 +1,313 @@
+    })
+
+    it.each([[`regex`, `/foo/`], [`glob`]])(
+      `does not support "%s" predicate`,
+      (predicate, value = `foo`) => {
+        const filter = {
+          a: { [predicate]: value },
+        }
+        expect(suggestIndex({ filter, sort })).toEqual([])
+      }
+    )
+
+    it.each([
+      [{ a: { in: 1 }, b: { eq: 1 }, c: { gt: 1 } }, [`b`, `a`, `c`]],
+      [{ a: { ne: 1 }, b: { eq: 1 }, c: { nin: [1, 2] } }, [`b`, `c`, `a`]],
+      [{ a: { glob: `*` }, b: { in: 1 }, c: { gt: 1 } }, [`b`, `c`]],
+      [
+        { a: { gt: 1 }, b: { gte: 1 }, c: { lte: 1 }, d: { lt: 1 } },
+        [`b`, `c`, `a`, `d`],
+      ],
+    ])(
+      `includes fields in index by predicate specificity: %#`,
+      (filter, expectedFields) => {
+        // expected specificity: eq, in, gte, lte, gt, lt
+        const expected = expectedFields.map(f => [f, 1])
+        expect(suggestIndex({ filter, sort })).toEqual(expected)
+      }
+    )
+
+    it(`adds at most 4 filter fields to index`, () => {
+      const filter = {
+        a: { eq: 1 },
+        b: { eq: 2 },
+        c: { eq: 3 },
+        d: { eq: 4 },
+        e: { eq: 5 },
+      }
+      // expected specificity: eq, in, gte, lte, gt, lt
+      expect(suggestIndex({ filter, sort })).toEqual([
+        [`a`, 1],
+        [`b`, 1],
+        [`c`, 1],
+        [`d`, 1],
+      ])
+    })
+  })
+
+  describe(`filter: no; sort: yes`, () => {
+    const filter = {}
+
+    it(`adds a single sort field to index`, () => {
+      const sort = { fields: [`a`], order: [] }
+      expect(suggestIndex({ filter, sort })).toEqual([[`a`, 1]])
+    })
+
+    it.each([
+      [[], 1],
+      [[`asc`], 1],
+      [[`ASC`], 1],
+      [[true], 1],
+      [[`desc`], -1],
+      [[`DESC`], -1],
+      [[false], -1],
+    ] as Array<[Sort["order"], number]>)(
+      `supports sort order defined as %p`,
+      (order, expected) => {
+        const sort = { fields: [`a`], order }
+        expect(suggestIndex({ filter, sort })).toEqual([[`a`, expected]])
+      }
+    )
+
+    it(`adds multiple sort fields to index`, () => {
+      const sort = { fields: [`a`, `b`], order: [] }
+      expect(suggestIndex({ filter, sort })).toEqual([
+        [`a`, 1],
+        [`b`, 1],
+      ])
+    })
+
+    it(`adds at most 4 sort fields to index`, () => {
+      const sort = {
+        fields: [`a`, `b`, `c`, `d`, `e`],
+        order: [],
+      }
+      expect(suggestIndex({ filter, sort })).toEqual([
+        [`a`, 1],
+        [`b`, 1],
+        [`c`, 1],
+        [`d`, 1],
+      ])
+    })
+
+    it(`only sorts in one direction`, () => {
+      // This is a limitation of our implementation :/
+      const sort: Sort = {
+        fields: [`a`, `b`, `c`],
+        order: [`desc`, `DESC`, `asc`],
+      }
+      expect(suggestIndex({ filter, sort })).toEqual([
+        [`a`, -1],
+        [`b`, -1],
+      ])
+    })
+
+    it.todo(`supports mixed sort order`)
```

Comment:
blocked on https://github.com/DoctorEvidence/lmdb-store/discussions/62#discussioncomment-898949 ?

Reply:
Originally - yes, was blocked because of this. But also this `todo` only relates to sorting using an index. Fallback with in-memory sort still works, so queries with mixed sort order will work (just will be slow and memory hog for now).

P.S. I think `lmdb-store` 1.6.0 does key encoding in JS now. So technically we probably can already implement this. Just requires some research and understanding of ordered binary encoding.